feature_engine/discretisation/arbitrary.py

```diff
@@ -147,13 +147,17 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         return self
 
-    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
-        """Sort the variable values into the intervals.
+    def transform(self, X: pd.DataFrame, precision: int = 3) -> pd.DataFrame:
```

Comment:
why do we need this change?

Reply:
We have to increase precision to correctly calculate geometric bins with `pd.cut`. Its either add parameter to method of the base class or change the value inside from 3 to 7, which will impact all discretisers